gpkit/model.py

```diff
@@ -421,69 +422,75 @@ def feasibility(self,
-                result = prod_gp.solve(verbosity=verbosity-1)
-                con_infeas = [result["variables"][sv] for sv in slackvars]
-                if verbosity > 0:
-                    print("  constraints : %s" % con_infeas)
-                feasibilities["constraints"] = con_infeas
-
-        constants = get_constants(unsubbed, allsubs)
-        if constvars:
-            constvars = set(constvars)
-            # get varkey versions
-            constvars = get_constants(unsubbed.varkeys, unsubbed.varlocs,
-                                dict(zip(constvars, constvars)))
-            # filter constants
-            constants = {k: v for k, v in constants.items() if k in constvars}
-        if "constants" in search and constants:
-            slackb = VectorVariable(len(constants))
-            constvarkeys, constvars, rmvalue, addvalue = [], [], {}, {}
-            for vk in constants.keys():
-                descr = dict(vk.descr)
-                del descr["value"]
-                vk_ = VarKey(**descr)
-                rmvalue[vk] = vk_
-                addvalue[vk_] = vk
-                constvarkeys.append(vk_)
-                constvars.append(Variable(**descr))
-            constvars = PosyArray(constvars)
-            constvalues = PosyArray(constants.values())
-            constraints = [c.sub(rmvalue) for c in self.constraints]
-            # cost function could also be .sum(), self.cost would break ties
-            var_m = Model(slackb.prod(),
-                          constraints
-                          + [slackb >= 1,
-                             constvalues/slackb <= constvars,
-                             constvars <= constvalues*slackb])
-            sol = var_m.solve(verbosity=verbosity-1)
-            feasible_constvalues = sol(constvars).tolist()
-            changed_vals = feasible_constvalues != np.array(constvalues)
-            var_infeas = {addvalue[constvarkeys[i]]: feasible_constvalues[i]
-                          for i in np.where(changed_vals)}
-            if verbosity > 0:
-                print("    constants : %s" % var_infeas)
-            feasibilities["constants"] = var_infeas
+        try:
+            m.gp()
+        except ValueError as err:
+            if err.message == ("GeometricPrograms cannot contain Signomials"):
+                raise ValueError("""Signomials remained after substitution.
 
-        if verbosity > 0:
-            print("")
+    'Model.feasibility()' can only be called on Models without Signomials,
+    because only those Models guarantee global feasibilities. Models with
+    Signomials have only local feasibilities, which can be found with
+    'Model.localfeasibility()'.""")
+            raise
+        else:
+            return self._feasibility(search, constvars, verbosity)
+
+    def localfeasibility(self,
+                         search=["overall", "constraints", "constants"],
+                         constvars=None, verbosity=0):
```

Comment:
It's unclear to me what the `search` arg does. This probably means this needs a docstring...